packages/create-pantheon-decoupled-kit/src/templates/next-drupal-app/app/layout.tsx

```diff
@@ -1,16 +1,27 @@
-import { Inter } from 'next/font/google';
-import './globals.css';
+'use client';
```

Comment:
I'm still wrapping my head around React Server Components, but I'm not sure we want this to be a client component.

From the docs (https://nextjs.org/docs/getting-started/react-essentials#client-components):

> Once "use client" is defined in a file, all other modules imported into it, including child components, are considered part of the client bundle.

So to my understanding, the `use client` directive on the layout would make the entire app client side rendered since this component is at the top of the tree.
We want layout to be a server component since it will include fetching for the footer menu at some point soon.

Reply:
Oh yeah I was trying to avoid making this a client component but it was required with the use of the `nextjs-kit` header. Let me look into work arounds, thanks for the docs ref!

Reply:
> it was required with the use of the nextjs-kit header.

Hmm that is a red flag to me. I dug into this and it looks like the nextjs-kit bundle is still a bit off, but when testing the correct bundle locally I kept running into this issue: https://github.com/vercel/next.js/issues/46267

The reason next is complaining is because the bundle is not correct, the file with the component also has hooks which are client only. So the error makes sense in that case, but when we split everything out it should be able to use only the Header component (which should work on the server).
 
I can open a draft PR to your branch with the bundle fix which we should do eventually, but it is not having much effect right now.

Reply:
Also don't have enough legit experience with RSC yet, but conceptually I actually can't think of anything that would need to be a client component in this new architecture. Am I missing something?

Reply:
> I actually can't think of anything that would need to be a client component in this new architecture

Anything with a hook will need to be a client component. We are really only using hooks (`useRouter`, `useEffect`) for the "go back" link and for the paginator component. I think it's fine if those are client components and we can rearchitect them if needed. Unfortuently, next.js doesn't seem to understand our bundle and when it tries to import a component, it does so from the `index.mjs` bundle which includes ALL the components, so next thinks it's a client component because `useEffect` is imported to that file, even though it is not used in the `Header` component..

Reply:
Agh that is unfortunate, thanks for digging into this @CobyPear. I have a few questions about this to bring up during standup.

Reply:
> Anything with a hook will need to be a client component.

Good callout. As I understand it, it isn't all hooks that can't be run on the server, but the rules are broad enough that it ends up being most of them. Based on the reliance on hooks I think this is going to break most people's brains when they migrate to server components.

> We are really only using hooks (useRouter, useEffect) for the "go back" link and for the paginator component.

With the app directory, I've been assuming that it would be possible to implement pagination server side.

And given this restriction, I wonder about the value of the 'go back' link. Would we just be better off with more traditional breadcrumbs that don't rely on routing?